### PR TITLE
Gun Smoke Particles

### PR DESCRIPTION
**Pull request by tyeagg**
## About The Pull Request
when you fire gun smoke come out

https://github.com/tgstation/TerraGov-Marine-Corps/assets/98577688/41aa6b1d-5af0-40c6-9d37-4d3c79c33dee
## Why It's Good For The Game
looks cool
## Changelog
:cl:
add: smoke particles for gun firing
/:cl:

### CODE REVIEW

**Comment by novaepee** [comment]
TGMC: The Woo Edition

tumblr_n2v3gbDdnT1sgrxw6o1_500-4208352554

**Comment by yyzsong** [comment]
Please do not make firefights even more of a headache than they usually are.

**Review by Kuro020 — APPROVED** [review]
Speedmerge or @TiviPlus is cringe

File: code/modules/projectiles/guns/grenade_launchers.dm [hunk 1]
```diff
@@ -222,7 +222,7 @@ The Grenade Launchers
 	fire_sound = 'sound/weapons/guns/fire/flare.ogg'
 	fire_sound = 'sound/weapons/guns/fire/flare.ogg'
 	w_class = WEIGHT_CLASS_SMALL
-	flags_gun_features = NONE
+	flags_gun_features = NONE|GUN_SMOKE_PARTICLES
```
Comment (Lumipharon) [inline code/modules/projectiles/guns/grenade_launchers.dm:225]: bruh

**Review by Lumipharon — CHANGES_REQUESTED** [review]
Why do energy weapons have this?

**Comment by Lumipharon** [comment]
It looks weird if you're shooting while moving, since the player sprite moves smoothly but the gun smoke will appear on their actual tile (i.e. the gun smoke will appear in front of you).

**Comment by AzonStarfall** [comment]
Looks really nice, but how will this hold up stability-wise when we have 50+ marines on a highpop round all spraying additional particles? Would the effects be minimal?

**Comment by iforgotmeword** [comment]
is this server sided?
if yes: time dilation goes BRRRRRRRRRRRRRRR

**Comment by tyeagg** [comment]
> is this server sided? if yes: time dilation goes BRRRRRRRRRRRRRRR

particles are like the very few things that byond has rendered clientside and also multicore, the only thing server side is the objects that hold the particle

**Comment by lbnesquik** [comment]
Test merge for performance

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**